### ISSUE TITLE
URL terminology coherence

### ISSUE THREAD

**Issue by jfoclpf** [jfoclpf/autocosts#117]
URL: http://video.google.co.uk:80/videoplay?docid=-7246927612831078230&hl=en#00h02m30s 

 - The protocol is http. Other protocols include https, ftp, etc.
 - The host or hostname is video.google.co.uk.
 - The subdomain is video.
 - The domain name is google.co.uk.
 - The top-level domain or TLD is uk. The uk domain is also referred to as a country-code top-level domain or ccTLD. For google.com, the TLD would be com.
 - The second-level domain (SLD) is co.uk.
 - The port is 80, which is the default port for web servers. Other ports are possible; a web server can listen on port 8000, for example. When the port is 80, most people leave out the port.
 - The path is /videoplay. Path typically refers to a file or location on the web server, e.g. /directory/file.html
 - This URL has parameters. The name of one parameter is docid and the value of that parameter is 7246927612831078230. URLs can have lots parameters. Parameters start with a question mark (?) and are separated with an ampersand (&).

**Comment by jfoclpf** [comment]
Nonetheless how Google calls it, 

we should follow RFC 6454 section 4 used by Javascript on browsers on  `window.location`

```
protocol://username:password@hostname:port/pathname?search#hash
-----------------------------href------------------------------
                             -----host----
-----------      origin      -------------
```

 - `protocol` - protocol scheme of the URL, including the final ':'
 - `hostname` - domain name
 - `port` - port number
 - `pathname` - /pathname
 - `search` - ?parameters
 - `hash` - #fragment_identifier
 - `username` - username specified before the domain name
 - `password` - password specified before the domain name
 - `href` - the entire URL
 - `origin` - protocol://hostname:port
 - `host` - hostname:port

Formal definition is in RFC 6454 section 4.
from: https://stackoverflow.com/a/28049503/3314008

**Comment by jfoclpf** [comment]
Check https://stackoverflow.com/questions/54722548/url-parts-canonical-terminology

**Comment by jfoclpf** [comment]
solved with 2515bb0935bc28bd85d348a8b6b135b554ae6449
